### PR TITLE
feat(ui): add countdown to cronWorkflowList Closes #4636

### LINKED ISSUE #4636: UI cron workflow countdown column
# Summary

What change needs making?

Would having a countdown column be something nice to add?

# Use Cases

When would you use this?

We have teams working at different timezones and with the addition of DST it's not straightforward to tell from the cron which workflow will execute next

image

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by tczhao**
Signed-off-by: Tianchu Zhao <evantczhao@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

Closes #4636

image

### CODE REVIEW

File: ui/src/app/shared/cron.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import parser = require('cron-parser');
```
Comment (simster7) [inline ui/src/app/shared/cron.tsx:1]: I like that you put this on its own file

**Review by simster7 — CHANGES_REQUESTED** [review]
Minor comments

File: ui/src/app/shared/components/duration-panel.tsx [hunk 2]
```diff
@@ -17,3 +18,21 @@ export const DurationPanel = (props: {phase: NodePhase; duration: number; estima
...
+            setDate(getDate);
+        }, frequency);
+        return () => {
+            clearInterval(interval);
+        };
+    }, []);
+
+    const duration = moment.duration(moment(date).diff(now)).asSeconds();
+
+    return <>{formatDuration(duration)}</>;
```
Comment (alexec) [inline ui/src/app/shared/components/duration-panel.tsx:37]: could you use <TIcker/> for this?

Comment (tczhao) [reply ui/src/app/shared/components/duration-panel.tsx:37]: Sorry, I don't quite understand the comment,
would you mind providing me more detail

Comment (alexec) [reply ui/src/app/shared/components/duration-panel.tsx:37]: There is a component called <Ticker/> that I think does this

Comment (tczhao) [reply ui/src/app/shared/components/duration-panel.tsx:37]: There's `<Moment duration={now} date={date} />`.
I thought we want to keep the formatting the same, thus using `formatDuration`

**Review by simster7 — APPROVED** [review]

**Pull request merged**